File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/currencyproperties.rb
module Asposetasksjava
  module CurrencyProperties
    def initialize()
        # Reading Currency Properties
        get_currency_properties()

        # Writing Currency Properties 
        set_currency_properties()
    end

    def get_currency_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        prj = Rjb::import('com.aspose.tasks.Prj')

        # Display currency properties
        puts "Currency Code : " + project.get(prj.CURRENCY_CODE).toString()
        puts "<br>Currency Digits : " + project.get(prj.CURRENCY_DIGITS).toString()
        puts "<br>Currency Symbol : " + project.get(prj.CURRENCY_SYMBOL).toString()
        puts "Currency Symbol Position: " + project.get(prj.CURRENCY_SYMBOL_POSITION).toString()
    end

    def set_currency_properties()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new

        prj = Rjb::import('com.aspose.tasks.Prj')

        # Set currency properties
        project.set(prj.CURRENCY_CODE, "AUD")
        project.set(prj.CURRENCY_DIGITS, 2)
        project.set(prj.CURRENCY_SYMBOL, "$")
        project.set(prj.CURRENCY_SYMBOL_POSITION, Rjb::import('com.aspose.tasks.CurrencySymbolPositionType').After)

        project.save(data_dir + "currency_properties.xml", Rjb::import('com.aspose.tasks.SaveFileFormat').XML)

        puts "Set currency properties, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/CurrencyProperties.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Prj as Prj;
use com\aspose\tasks\CurrencySymbolPositionType as CurrencySymbolPositionType;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

class CurrencyProperties{

    public static function run($dataDir=null){

        # Reading Currency Properties
        CurrencyProperties::get_currency_properties($dataDir);

        # Writing Currency Properties
        CurrencyProperties::set_currency_properties($dataDir);
    }

    public static function get_currency_properties($dataDir=null){
        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $prj = new Prj();

        # Display currency properties
        print "Currency Code : " . (string)$project->get($prj->CURRENCY_CODE).PHP_EOL;//.toString()
        print "<br>Currency Digits : " . (string)$project->get($prj->CURRENCY_DIGITS).PHP_EOL;//.toString()
        print "<br>Currency Symbol : " . (string)$project->get($prj->CURRENCY_SYMBOL).PHP_EOL;//.toString()
        print "Currency Symbol Position: ". (string)$project->get($prj->CURRENCY_SYMBOL_POSITION).PHP_EOL;//.toString()
    }

    public static function set_currency_properties($dataDir=null){

        # Instantiate project object
        $project = new Project();

        $prj = new Prj();

        # Set currency properties
        $currencySymbolPositionType=new CurrencySymbolPositionType();

        $project->set($prj->CURRENCY_CODE, "AUD");
        $project->set($prj->CURRENCY_DIGITS, 2);
        $project->set($prj->CURRENCY_SYMBOL, "$");
        $project->set($prj->CURRENCY_SYMBOL_POSITION, $currencySymbolPositionType->After);

        $saveFileFormat=new SaveFileFormat();
        $project->save($dataDir . "currency_properties.xml", $saveFileFormat->XML);
        print "Set currency properties, please check the output file.".PHP_EOL;
    }

}
?>